site/content/how-to/monitoring/troubleshooting.md

```diff
@@ -65,54 +65,350 @@ LAST SEEN   TYPE      REASON              OBJECT
+
+#### Modify log levels before deploying
+
+1. If using manifests, edit `deploy/manifests/nginx-gateway.yaml` to update the logging level for `nginx-gateway-config`:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: NginxGateway
+metadata:
+  name: nginx-gateway-config
+  namespace: nginx-gateway
+  labels:
+    app.kubernetes.io/name: nginx-gateway
+    app.kubernetes.io/instance: nginx-gateway
+    app.kubernetes.io/version: "edge"
+spec:
+  logging:
+    level: debug
+```
+
+1. If using helm, add `--set nginxGateway.config.logging.level=<log-level>` to your helm installation command.
+
+#### Modify log levels after deploying
+
+Once you have deployed NGINX Gateway Fabric, you can modify log levels by editing the config for NGINX Gateway as shown below:
+
+```shell
+kubectl [-n namespace] edit nginxgateways ngf-config
+```
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: NginxGateway
+metadata:
+  annotations:
+    meta.helm.sh/release-name: ngf
+    meta.helm.sh/release-namespace: nginx-gateway
+  creationTimestamp: "2024-06-12T18:35:05Z"
+  generation: 1
+  labels:
+    app.kubernetes.io/instance: ngf
+    app.kubernetes.io/managed-by: Helm
+    app.kubernetes.io/name: nginx-gateway-fabric
+    app.kubernetes.io/version: edge
+    helm.sh/chart: nginx-gateway-fabric-1.3.0
+  name: ngf-config
+  namespace: nginx-gateway
+  resourceVersion: "62293"
+  uid: fa6d6a12-14e1-4168-95d5-595e7f63b270
+spec:
+  logging:
+    level: debug
+```
+
 ### NGINX fails to reload
 
 #### Description
 
+NGINX reload errors can occur for various reasons, including syntax errors in configuration files, permission issues, and more. To determine if NGINX has failed to reload, check logs for your _nginx-gateway_ and _nginx_ containers.
+You will see the following error in the _nginx-gateway_  logs `failed to reload NGINX:` followed by the reason for the failure. Similarly, you will see error logs in the _nginx_ container as `2024/06/12 14:25:11 [emerg] 12345#0: open() "/var/run/nginx.pid" failed (13: Permission denied)`.
+
+To debug why your reload has failed, start with verifying the syntax of your configuration files by opening a shell in the NGINX container following these [steps](#get-shell-access-to-containers) and running `nginx -T`. If there are errors in your configuration file, the reload will fail and specify why it has failed.
+
+### Understanding the generated config
+
+Understanding the NGINX configuration is key for fixing issues because it shows how NGINX handles requests. This helps tweak settings to make sure NGINX behaves the way you want it to for your application. The configuration file is found at /etc/nginx/nginx.conf within your NGINX Container. To understand the usage of NGINX Directives in the configuration file, consult this list of [NGINX Directives](https://nginx.org/en/docs/dirindex.html).
```

Comment:
```suggestion
Understanding the NGINX configuration is key for fixing issues because it shows how NGINX handles requests. This helps tweak settings to make sure NGINX behaves the way you want it to for your application. The configuration file is found at /etc/nginx/nginx.conf within your NGINX container. To understand the usage of NGINX directives in the configuration file, consult this list of [NGINX directives](https://nginx.org/en/docs/dirindex.html).
```